### ISSUE TITLE
inversion: susceptibility tensor imaging (STI)

### ISSUE THREAD

**Issue by astewartau** [astewartau/QSM.rs#113]
## Summary

Add susceptibility tensor imaging — recover the 6-component symmetric susceptibility tensor from field maps at >= 6 (in practice ~12) orientations. Liu C, *MRM* 2010;63(6):1471–7. Review: Li W et al., *NMR Biomed* 2017 (<https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/abs/10.1002/nbm.3540>).

**This supersedes the LOW PRIORITY entry in `TODO.md`**, which was based on a wrong premise (see below).

## Correcting the TODO entry

`TODO.md` currently lists STI as LOW PRIORITY, citing `Core_Functions_P/STI_Parfor.m` as a "compiled .p". That is wrong on two counts:

1. **`STI_Parfor.m` is plain readable MATLAB source**, not obfuscated. Unlike most of STI Suite, there is no `STI_Parfor.p` — the `.m` is the implementation.
2. There is a **second, independent reference implementation** with test data: `apply_STI.m` in Bilgic's `COSMOS_STI_Toolbox.zip` (see #112).

With two readable references and 12-orientation data in hand, this is considerably smaller than the TODO implies.

## Key observation: the two references solve the same system, and the direct route is simpler

Bilgic's `apply_STI.m` builds a 6-channel forward/adjoint operator and runs global LSQR over it. STI Suite's `STI_Parfor.m` does a per-k-voxel direct solve (`A\thetai`).

Reading `apply_STI.m`: the operator is **purely pointwise in k**. It couples the 6 tensor channels to the N orientations at each k-point with no spatial coupling whatsoever. So the global LSQR system is **block-diagonal in k** — it decomposes into `N_k` independent 6-unknown least-squares problems. STI Suite's per-voxel solve is the *exact* solution; Bilgic's LSQR is an early-stopped approximation of it.

**Recommendation: implement the per-voxel direct solve.** With 12 orientations each system is 12x6 overdetermined, so 6x6 normal equations plus Cholesky per k-voxel. Exact, embarrassingly parallel with rayon, and no iteration count to tune.

Optionally add the LSQR path as well for the implicit regularisation that early stopping provides on noisy data — `inversion::ilsqr::lsqr` (`src/inversion/ilsqr.rs:62`) is already generic over closures, so it is nearly free to wire up.

## Forward model

Per orientation `t`, at each k-point:

```
delta_t(k) = (1/3) . H_t^T chi(k) H_t  -  (H_t . k)(k^T chi(k) H_t) / k^2
```

6 unknowns per k-point: `[chi11, chi12, chi13, chi22, chi23, chi33]`.

Handle the `k = 0` singularity — both references guard with `eps`; STI Suite additionally maps `NaN -> 0`.

## Post-processing is already covered by existing code

- **MMS** (mean magnetic susceptibility) = mean of eigenvalues
- **MSA** (magnetic susceptibility anisotropy) = `lambda_3 - (lambda_1 + lambda_2)/2`

Both need eigenvalues only: `eigenvalues_3x3_cardano` (`src/utils/frangi.rs:355`) covers it.

- **Principal eigenvector** (fibre orientation) needs a full eigendecomposition: `src/utils/denoise.rs:23` already has a symmetric eigensolver returning eigenvalues **and** eigenvectors, ascending order, column-wise.

No new numerics required.

## Units

Input: **local field in ppm**, matching the crate convention. Note STI Suite's `STI_Parfor` instead takes raw radians and applies `ScalingFactor(B0, TE_ms)` at the end; Bilgic's takes normalised field directly. We follow Bilgic / the crate convention.

## Testing

The 12-orientation dataset from `COSMOS_STI_Toolbox.zip` (160^3 x 12 field + `R_tot` + mask) validates this end-to-end against the published tensor-component and MMS/MSA figures.

For a cheap unit test, the k-space block-diagonal structure means a single k-point can be tested in isolation: build a known tensor, forward-project to N orientations, solve, assert recovery.

Note on synthetic data: `qsm-forward` has `generate_theta_from_v1` (fibre orientation V1 -> orientation-dependent susceptibility), but as its own `prototypes/hollow_cylinder` notes point out, it simulates an orientation-modulated **scalar** chi, not a full tensor. It is therefore not a substitute for real STI ground truth.

## Effort

~350 lines: operator + per-voxel solve + MMS/MSA/PEV post-processing.

## Follow-up (blocked upstream): DeepSTI

DeepSTI (Fang et al., *Med Image Anal* 2023, <https://www.sciencedirect.com/science/article/pii/S1361841523000890>) reconstructs tensors from **fewer than 6** orientations via a learned proximal operator, which would fit this crate's existing ONNX infrastructure well.

Checked <https://github.com/ZhenghanFang/DeepSTI-pytorch>: **no pretrained weights are published and there is no license file.** Blocked on upstream. Logging it here as a follow-up, not a task.

## Depends on

#111 (rigid registration) to align the orientations.

## Licensing

STI Suite is "free for academic use"; Bilgic's toolbox states no license. Both publish their equations in peer-reviewed work. Port from the published math, use the code only to resolve conventions.
